File: src/main/java/fr/rhaz/os/android/Menu.java
package fr.rhaz.os.android;

import android.Manifest;
import android.Manifest.permission;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Environment;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;

import java.io.File;

import fr.rhaz.os.android.plugins.PluginsActivity;

public class Menu extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        int permissions_code = 42;
        String[] permissions = {permission.WRITE_EXTERNAL_STORAGE, permission.READ_EXTERNAL_STORAGE};

        ActivityCompat.requestPermissions(this, permissions, permissions_code);
    }

    public Intent intent(Class<?> cls){
        return new Intent(this, cls);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults){
        for(int r:grantResults)
            if(r != PackageManager.PERMISSION_GRANTED) return;

        File folder = new File(Environment.getExternalStorageDirectory(), "RHazOS");
        try {
            if(!folder.exists())
                if(!folder.mkdirs())
                    Log.w("RHazOS", "Could not create folder");
            folder.setWritable(true);
        } catch(Exception e){
            e.printStackTrace();
            return;
        }

        if (!Utils.isServiceRunning(this, ConsoleService.class))
            startService(intent(ConsoleService.class));

        findViewById(R.id.startbutton).setOnClickListener(v -> startActivity(intent(ConsoleActivity.class)));
        findViewById(R.id.pluginsbutton).setOnClickListener(v -> startActivity(intent(PluginsActivity.class)));
        findViewById(R.id.optionsbutton).setOnClickListener(v -> Utils.openFolder(Menu.this, new File(Environment.getExternalStorageDirectory(), "RHazOS")));
        findViewById(R.id.aboutbutton).setOnClickListener(v -> startActivity(intent(AboutActivity.class)));
    }
}

File: src/main/java/fr/rhaz/os/android/Menu.kt
package fr.rhaz.os.android

import android.Manifest.permission
import android.content.pm.PackageManager
import android.os.Bundle
import android.os.Environment
import android.support.v4.app.ActivityCompat
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View

import java.io.File

import fr.rhaz.os.android.plugins.PluginsActivity
import org.jetbrains.anko.intentFor
import org.jetbrains.anko.startActivity

class Menu : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        val requestCode = 42
        val permissions = arrayOf(permission.WRITE_EXTERNAL_STORAGE, permission.READ_EXTERNAL_STORAGE)

        ActivityCompat.requestPermissions(this, permissions, requestCode)
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        for (r in grantResults)
            if (r != PackageManager.PERMISSION_GRANTED) return

        val folder = File(Environment.getExternalStorageDirectory(), "RHazOS")
        try {
            if (!folder.exists())
                if (!folder.mkdirs())
                    Log.w("RHazOS", "Could not create folder")
            folder.setWritable(true)
        } catch (e: Exception) {
            e.printStackTrace()
            return
        }

        if (!Utils.isServiceRunning(this, ConsoleService::class.java))
            startService(intentFor<ConsoleService>())

        findViewById<View>(R.id.startbutton).setOnClickListener { startActivity<ConsoleActivity>() }
        findViewById<View>(R.id.pluginsbutton).setOnClickListener { startActivity<PluginsActivity>() }
        findViewById<View>(R.id.optionsbutton).setOnClickListener { Utils.openFolder(it,this@Menu, File(Environment.getExternalStorageDirectory(), "RHazOS")) }
        findViewById<View>(R.id.aboutbutton).setOnClickListener { startActivity<AboutActivity>() }
    }
}
